src/mono/cmake/configure.cmake

```diff
@@ -155,6 +155,20 @@ check_c_source_compiles(
   "
   HAVE_GNU_STRERROR_R)
 
+if (HOST_FREEBSD)
+check_c_source_compiles(
```

Comment:
nit: we can replace `check_c_source_compiles` with a one liner `check_symbol_exists` to replace both if/else branches (see its usage in this file and other places):

```cmake
check_symbol_exists(CPU_COUNT "sched.h" HAVE_GNU_CPU_COUNT)
```

Reply:
That is a much better way of doing it